lms/views/basic_lti_launch.py

```diff
@@ -105,23 +116,43 @@ def db_configured_basic_lti_launch(self):
         in the LMS and passing it back to us in each launch request. Instead we
         retrieve the document URL from the DB and pass it to Via.
         """
-        self.sync_lti_data_to_h()
-        self.store_lti_data()
-        self.course_service.get_or_create(self.context.h_group.authority_provided_id)
-
-        self.context.js_config.maybe_enable_grading()
-
         # The ``db_configured=True`` view predicate ensures that this view
         # won't be called if there isn't a matching document_url in the DB. So
         # here we can safely assume that the document_url exists.
-        self.context.js_config.add_document_url(
-            self.assignment_service.get_document_url(
-                self.request.params["tool_consumer_instance_guid"],
-                self.request.params["resource_link_id"],
-            )
+        tool_consumer_instance_guid = self.request.params["tool_consumer_instance_guid"]
+        resource_link_id = self.request.params["resource_link_id"]
+        document_url = self.assignment_service.get_document_url(
+            tool_consumer_instance_guid, resource_link_id
         )
+        return self.basic_lti_launch(document_url)
 
-        return {}
+    @view_config(blackboard_copied=True)
+    def blackboard_copied_basic_lti_launch(self):
+        """
+        Respond to a launch of a newly-copied Blackboard assignment.
+
+        Find the document_url for resource_link_id_history and make a copy of
+        it with the new resource_link_id, then lanuch the assignment as normal.
+
+        For more about Blackboard course copy and the resource_link_id_history
+        param see the BlackboardCopied predicate's docstring.
+        """
+        # The ``blackboard_copied=True`` view predicate ensures that this view
+        # won't be called if there isn't a resource_link_id_history param and a
+        # matching document_url in the DB.
+        tool_consumer_instance_guid = self.request.params["tool_consumer_instance_guid"]
+        resource_link_id = self.request.params["resource_link_id"]
+        resource_link_id_history = self.request.params["resource_link_id_history"]
+
+        document_url = self.assignment_service.get_document_url(
+            tool_consumer_instance_guid, resource_link_id_history
+        )
+
+        self.assignment_service.set_document_url(
+            tool_consumer_instance_guid, resource_link_id, document_url
+        )
+
+        return self.basic_lti_launch(document_url)
```

Comment:
This is the new view method for a new type of assignment launch: a "Blackboard copied" assignment launch. The predicates (below) will direct launch requests to this view as appropriate